ui/client/reducers/settings.ts

```diff
@@ -33,16 +33,23 @@ export type SettingsState = {
 }
 
 export type BaseAuthenticationSettings = {
-  backend?: string
+  provider?: string
+  strategy?: string
 }
 
-export type AuthenticationSettings = BaseAuthenticationSettings | RemoteAuthenticationSettings | OAuth2Settings
+export type AuthenticationSettings = BaseAuthenticationSettings | BrowserAuthenticationSettings | RemoteAuthenticationSettings | OAuth2Settings
```

Comment:
why do we need these BaseAuthenticationSettings here?

Reply:
To allow nulls for the provider and strategy fields in the initial state.